cube/src/main/java/zone/cogni/asquare/cube/sort/SortedStatementsString.java

```diff
@@ -16,35 +17,98 @@
 
 public class SortedStatementsString implements Function<List<Statement>, String> {
 
-  private static final int DEFAULT_INDENT = 24;
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  public static class Builder {
+    private String base;
+    private Map<String, String> namespaces;
+    private int indent = 8;
+
+    public Builder withBase(String base) {
+      this.base = base;
+      return this;
+    }
+
+    public Builder withNamespaces(Map<String, String> namespaces) {
+      this.namespaces = namespaces;
+      return this;
+    }
+
+    public Builder withIndent(int indent) {
+      if (indent < 0) throw new RuntimeException("cannot have negative indent");
+
+      this.indent = indent;
+      return this;
+    }
+
+    public SortedStatementsString build() {
+      SortedStatementsString result = new SortedStatementsString();
+
+      if (StringUtils.isNotBlank(base))
+        result.setBase(base);
+
+      if (MapUtils.isNotEmpty(namespaces))
+        result.setNamespaces(namespaces);
+
+      result.setIndent(indent);
+
+      return result;
+    }
+
+  }
 
-  private final int indent;
+  private String base;
   private final Map<String, String> namespaces = new TreeMap<>();
+  private int indent;
 
   public SortedStatementsString() {
-    this(DEFAULT_INDENT, ImmutableMap.of());
   }
 
-  public SortedStatementsString(Map<String, String> namespaces) {
-    this(DEFAULT_INDENT, namespaces);
+  public String getBase() {
+    return base;
   }
 
-  public SortedStatementsString(int indent, Map<String, String> namespaces) {
-    this.indent = indent;
+  public void setBase(String base) {
+    this.base = base;
+  }
+
+  public Map<String, String> getNamespaces() {
+    return namespaces;
+  }
+
+  public void setNamespaces(Map<String, String> namespaces) {
```

Comment:
should we make a diff between setNamespaces and addNamespaces
this looks actually like add, not set

Reply:
hmm. true. this is only true in case of use of builder and we start with an empty map.